gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -480,6 +480,17 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
         {{ "\n" }}
     {% endfor %}
 
+    def __enter__(self):
+        self.transport.__enter__()
+        return self
+
+    def __exit__(self, type, value, traceback):
```

Comment:
I think we need a stronger warning, something on the lines of "ONLY use as a context manager if the transport is NOT shared with other clients! Exiting the with block will CLOSE the transport and may cause errors in other clients!"

Reply:
Done.